### ISSUE TITLE
Catching Websocket errors

### ISSUE THREAD

**Issue by andrewdavidcostello** [twilio/twilio-video.js#289]
We are going to be using Twilio Video in locations with very strict firewall setups, as such we need to know when it is a firewall issue to Twilio.

I can't seem to find anywhere that gives you a hook into capturing an error with the websocket connection for any endpoint but for this case the output was:

`WebSocket connection to 'wss://endpoint.twilio.com/' failed: Error in connection establishment: net::ERR_CERT_COMMON_NAME_INVALID`

Is there any way to pass in an option to hook into the WS, at least for the error?

I have tried the usual try..catch, window.onerror but nothing seems to work outside of it.

Any pointers would be much appreciated.

**Comment by manjeshbhargav** [comment]
@andrewdavidcostello ,

You can get access to an error while connecting to a Room as follows:

```javascript
Twilio.Video.connect({ room: 'my-room' }).catch(error => {...});
```

Please refer to the documentation for [connect](https://media.twiliocdn.com/sdk/js/video/releases/1.8.0/docs/global.html#connect) for details.

Let me know if this works for you.

Thanks,

Manjesh Malavalli
SDK Platform Engineering

**Comment by andrewdavidcostello** [comment]
This only seems to catch errors dealing with the actual initial room connection (for instance an incorrect auth token), it does not seem to give any error relating the websocket which seems to occur after the initial connect or somewhere inbetween?

**Comment by manjeshbhargav** [comment]
@andrewdavidcostello ,

You can listen to the Room's "disconnected" event like so:

```javascript
const room = await Twilio.Video.connect(...);
room.on('disconnected', (room, error) => {
  if (error) {
    console.error(error);
  }
});
```

**Comment by mjamal-tiu** [comment]
Hello Team,

I am getting below error in console while I am trying to comment to Room. Please let me know how can I resolve or is it error at the twilio-video.js file which I am using from CDN.

WebSocket connection to 'wss://sdkgw.us1.twilio.com/v1/VideoEvents' failed: WebSocket is closed before the connection is established.

**Comment by manjeshbhargav** [comment]
Hi @mjamal-tiu ,

Can you share the full browser console logs? I'm interested to know whether you were able to connect to the Room in spite of this error.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by mjamal-tiu** [comment]
Issue is resolved based on suggestion provided on #576
